### PR TITLE
etcdserver: add >= support for v3 delete range

### PR DESCRIPTION
**Pull request by heyitsanthony**

### CODE REVIEW

File: clientv3/integration/kv_test.go [hunk 1]
```diff
@@ -229,35 +229,65 @@ func TestKVDeleteRange(t *testing.T) {
...
+			opts: []clientv3.OpOption{clientv3.WithFromKey()},
+
+			wkeys: []string{},
+		},
 	}
 
 	for i, tt := range tests {
-		dresp, err := kv.Delete(ctx, tt.key, clientv3.WithRange(tt.end))
+		keySet := []string{"a", "b", "c", "c/abc", "d"}
+		for i, key := range keySet {
```
Comment (hongchaodeng) [inline clientv3/integration/kv_test.go]: use 'j'

File: clientv3/integration/kv_test.go [hunk 2]
```diff
@@ -229,35 +229,65 @@ func TestKVDeleteRange(t *testing.T) {
...
+			if _, err := kv.Put(ctx, key, ""); err != nil {
+				t.Fatalf("#%d: couldn't put %q (%v)", i, key, err)
```
Comment (hongchaodeng) [inline clientv3/integration/kv_test.go]: use 'j'

**Comment by heyitsanthony** [comment]
fixed /cc @hongchaodeng

**Comment by heyitsanthony** [comment]
ok brokenness of delete confirmed...

**Comment by heyitsanthony** [comment]
fixed the delete command /cc @xiang90 @hongchaodeng

File: etcdserver/v3demo_server.go [hunk 3]
```diff
@@ -631,3 +633,9 @@ func compareInt64(a, b int64) int {
 		return 0
 	}
 }
+
+// isGteRange determins if the range end is a >= range. This works around grpc
+// sending empty byte strings as nil; >= is encoded in the range end as '\0'.
+func isGteRange(rangeEnd []byte) bool {
```
Comment (xiang90) [inline etcdserver/v3demo_server.go]: what is Gte?

**Comment by xiang90** [comment]
LGTM.

File: etcdserver/v3demo_server.go (hunk 3 above)
Comment (heyitsanthony) [inline etcdserver/v3demo_server.go]: Greater than or equal.

**Pull request merged**